src/Process/LayeredMaterialNodeProcessing.js

```diff
@@ -273,12 +286,21 @@ export function updateLayeredMaterialNodeImagery(context, layer, node) {
                     // eslint-disable-next-line no-console
                     console.warn(`Imagery texture update error for ${node}: ${err}`);
                 }
-                const definitiveError = node.layerUpdateState[layer.id].errorCount > MAX_RETRY;
-                node.layerUpdateState[layer.id].failure(Date.now(), definitiveError);
-                if (!definitiveError) {
-                    window.setTimeout(() => {
-                        context.view.notifyChange(false, node);
-                    }, node.layerUpdateState[layer.id].secondsUntilNextTry() * 1000);
+                const nextLevelToTry = targetLevel - 1;
+                const pFailureParams = node.layerUpdateState[layer.id].failureParams;
+                const pFailureLevel = pFailureParams ? pFailureParams.targetLevel : undefined;
+                // notify a new try specifying failure parameters
+                if (layer.tileInsideLimit(node, layer, nextLevelToTry) && nextLevelToTry != pFailureLevel) {
```

Comment:
I'd prefer not to have this logic here: it's a duplicate of the code you added 10 lines above.